src/test/kotlin/com/safehill/kclient/network/local/LocalServerInterfaceImpl.kt

```diff
@@ -33,49 +33,49 @@ class LocalServerInterfaceImpl : LocalServerInterface {
         publicKeyData: ByteArray,
         publicSignatureData: ByteArray,
     ) {
-        TODO("Not yet implemented")
+        error("Should not call this method")
     }
 
     override suspend fun createOrUpdateThread(threads: List<ConversationThreadOutputDTO>) {
-        TODO("Not yet implemented")
+        error("Should not call this method")
     }
 
     override suspend fun createOrUpdateThread(
         name: String?,
         recipientsEncryptionDetails: List<RecipientEncryptionDetailsDTO>,
     ): ConversationThreadOutputDTO {
-        TODO("Not yet implemented")
+        error("Should not call this method")
```

Comment:
Thread creation can be mocked. All these methods should be left as "not yet implemented". We should also implement them sooner than later so that we have some test coverage around thread creation, deletion, etc. using the local DB only.
You can look at the Swift counterpart (Safehill-Client) for inspiration.

For now, use:
- TODO("Not yet implemented") for things that can be tested on the client and are not
- Something like SafehillError.ServerError.NotSupported for the ones that doesn't make sense to test or implement in the LocalServer test implementation

Reply:
Oh got it..

Reply:
Refactored.

Reply:
Can we strongly type the error, like we do in Swift?

Reply:
The error handling has been refactored to mirror error handling in ios counterpart.